### PR TITLE
Feature (Add uid to db)

### PR DESCRIPTION
**Pull request by andrii1**
# Description:
Adding uid to users table

# Jira:

### CODE REVIEW

File: packages/client/src/containers/Signup/Signup.container.js [hunk 1]
```diff
@@ -26,12 +32,28 @@ function Signup() {
...
+        full_name: fullname,
+        email: userCreated.email,
+        uid: userCreated?.uid,
+      }),
+    };
+    await fetch(`${apiURL()}/users`, requestOptions)
+      .then((response) => response.json())
+      .then((data) => {
+        /* console.log(data);
+        console.log('user', userCreated, userCreated?.uid); */
```
Comment (orhantoy) [inline packages/client/src/containers/Signup/Signup.container.js]: What should happen here?

File: packages/client/src/containers/Signup/Signup.container.js [hunk 2]
```diff
@@ -26,12 +32,28 @@ function Signup() {
...
+    const requestOptions = {
+      method: 'POST',
+      headers: { 'Content-Type': 'application/json' },
+      body: JSON.stringify({
+        full_name: fullname,
+        email: userCreated.email,
+        uid: userCreated?.uid,
+      }),
+    };
+    await fetch(`${apiURL()}/users`, requestOptions)
```
Comment (orhantoy) [inline packages/client/src/containers/Signup/Signup.container.js]: Is there a particular reason for mixing await/async with `.then` below?

File: packages/client/src/containers/Signup/Signup.container.js [hunk 3]
```diff
@@ -46,8 +68,12 @@ function Signup() {
   };
   useEffect(() => {
     if (loading) return;
-    if (user) navigate('/');
-  }, [user, loading, navigate]);
+    if (user) {
+      /* console.log('user1', user, user.email); */
```
Comment (orhantoy) [inline packages/client/src/containers/Signup/Signup.container.js]: Let's remember to clean up commented code.

File: packages/server/api/controllers/users.controller.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+  }
+};
+
+// post
+const createUsers = async (body) => {
+  try {
+    await knex('users').insert({
+      full_name: body.full_name,
+      email: body.email,
+      uid: body.uid,
```
Comment (orhantoy) [inline packages/server/api/controllers/users.controller.js:9]: Did you figure out a way to deal with duplicate uids? Couldn't that happen if someone tries to sign up twice with the same email?

File: packages/server/api/routes/users.router.js [hunk 5]
```diff
@@ -27,4 +27,47 @@ router.get('/', (req, res, next) => {
     .catch(next);
```
Comment (orhantoy) [inline packages/server/api/routes/users.router.js]: I don't think this is needed so I would remove this route and the `getUsers` controller function.

File: packages/server/api/routes/users.router.js [hunk 6]
```diff
@@ -27,4 +27,47 @@ router.get('/', (req, res, next) => {
...
+ *    parameters:
+ *      - in: body
+ *        name: users
+ *        description: The user to create.
+ *        schema:
+ *          type: object
+ *          required:
+ *            - full_name
+ *            - email
+ *            - created_at
```
Comment (orhantoy) [inline packages/server/api/routes/users.router.js]: I don't think `created_at` is accepted as a parameter.

File: packages/server/api/routes/users.router.js [hunk 7]
```diff
@@ -27,4 +27,47 @@ router.get('/', (req, res, next) => {
...
+ *            - uid
+ *          properties:
+ *            full_name:
+ *              type: string
+ *            email:
+ *              type: string
+ *            created_at:
+ *              type: date/time
+ *            uid:
+ *              type: date/time
```
Comment (orhantoy) [inline packages/server/api/routes/users.router.js]: I think the type here should be string.

File: packages/server/api/routes/users.router.js [hunk 8]
```diff
@@ -27,4 +27,47 @@ router.get('/', (req, res, next) => {
...
+ *    responses:
+ *      201:
```
Comment (orhantoy) [inline packages/server/api/routes/users.router.js]: Does it actually return a 201 response?

File: packages/server/api/controllers/users.controller.js (hunk 4 above)
Comment (andrii1) [reply packages/server/api/controllers/users.controller.js:9]: it goes through firebase first, so user is not created if someone tries to sign up twice with the same email;

File: packages/client/src/containers/Signup/Signup.container.js (hunk 1 above)
Comment (andrii1) [reply packages/client/src/containers/Signup/Signup.container.js]: removed it

File: packages/client/src/containers/Signup/Signup.container.js (hunk 2 above)
Comment (andrii1) [reply packages/client/src/containers/Signup/Signup.container.js]: removed

File: packages/server/api/routes/users.router.js (hunk 8 above)
Comment (andrii1) [reply packages/server/api/routes/users.router.js]: changed to 200, or should it be removed?

File: packages/server/api/routes/users.router.js (hunk 5 above)
Comment (andrii1) [reply packages/server/api/routes/users.router.js]: ok, it was just for testing purposes

File: packages/server/api/routes/users.router.js (hunk 8 above)
Comment (orhantoy) [reply packages/server/api/routes/users.router.js]: It's fine with 200

**Review by orhantoy — APPROVED** [review]

**Pull request merged**